### ISSUE TITLE
Template literal types don't work correctly

### ISSUE THREAD

**Issue by MystPi** [DanielXMoore/Civet#441]
```civet
type Test<T> =
  T extends "(#{infer Rest}"
    ? [LParen, Rest]
    : T extends "#{infer Char extends Alpha}#{infer Rest}"
      ? [Character<Char>, Rest]
      : never
```
Becomes:
```ts
type Test<T> = T extends "(#{infer Rest}"
  ? [LParen, Rest]
  : T extends "#{infer Char extends Alpha}#{infer Rest}"
  ? [Character<Char>, Rest]
  : never;
```
While it should be:
```ts
type Test<T> = T extends `(${infer Rest}`
  ? [LParen, Rest]
  : T extends `${infer Char extends Alpha}${infer Rest}`
  ? [Character<Char>, Rest]
  : never;
```
Same thing with types like `"#{string}"`, `"#{number}"`, etc.

**Comment by STRd6** [comment]
This is fixed in v0.5.85. One note is that to use `"#{}"` as an interpolation you need to use the `"civet coffeeInterpolation"` directive.

**Comment by MystPi** [comment]
Awesome; thank you so much!
